contracts/LifCrowdsale.sol

```diff
@@ -102,39 +102,44 @@ contract LifCrowdsale is Ownable, PullPayment {
     // founders vesting compensation and presale payments
     function getMaxTokens() public returns (uint) {
       uint minTokenPrice = minCap.div(totalTokens);
-      uint maxDiscountedTokens = totalPresaleWei.div(minTokenPrice).mul(presaleDiscount.add(100).div(100));
-      uint maxFoundersTokens = maxDiscountedTokens.mul(ownerPercentage).div(1000);
+      uint maxWithBonusTokens = totalPresaleWei.div(minTokenPrice).mul(presaleBonusRate.add(100).div(100));
+      uint maxFoundersTokens = maxWithBonusTokens.mul(ownerPercentage).div(1000);
 
-      return totalTokens + maxDiscountedTokens + maxFoundersTokens;
+      return totalTokens + maxWithBonusTokens + maxFoundersTokens;
     }
 
-    // Add an address that would be able to spend certain amounts of ethers with discount
-    function addDiscount(address target, uint amount) external onlyOwner() onStatus(1,2) {
+    // Add an address that would be able to spend certain amounts of ethers with a bonus rate
+    function addPresalePayment(address target, uint amount) external onlyOwner() onStatus(1,2) {
 
-      if (presaleDiscount == 0)
+      if (presaleBonusRate == 0)
         throw;
 
       totalPresaleWei = totalPresaleWei.add(amount);
       presalePayments[target] = amount;
 
-      // check that crowdsale balance is AT LEAST max(discounted)tokens at lowest possible valuation +
+      // check that crowdsale balance is AT LEAST max(with bonus)tokens at lowest possible valuation +
       // founders tokens
       uint crowdsaleBalance = ERC20Basic(tokenAddress).balanceOf(address(this)).div(LONG_DECIMALS);
 
       if (crowdsaleBalance < getMaxTokens())
         throw;
     }
 
-    function distributeTokens(address buyer, bool discount) external onStatus(3, 0) {
+    function getBuyerPresaleTokens(address buyer) public returns (uint) {
```

Comment:
The only impact is having more gas consumption on the deployment of this contract and the other contracts that use it, but if its not to much and the function helps debugging and it makes the contract more readable we should keep it.